src/Aspire.Hosting/Dashboard/DashboardEventHandlers.cs

```diff
@@ -419,6 +419,31 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
             });
         }
 
+        dashboardResource.Annotations.Add(new ResourceUrlsCallbackAnnotation(c =>
+        {
+            foreach (var url in c.Urls)
+            {
+                switch (url.Endpoint?.EndpointName)
+                {
+                    case OtlpGrpcEndpointName:
+                        url.DisplayText = OtlpGrpcEndpointName;
+                        break;
+                    case OtlpHttpEndpointName:
+                        url.DisplayText = OtlpHttpEndpointName;
+                        break;
+                    default:
+                        // Other endpoints are for the dashboard UI. There are typically dashboard UI endpoints for http and https.
+                        // Order these before non-browser usable endpoints.
+                        if (Uri.TryCreate(url.Url, UriKind.Absolute, out var result))
+                        {
+                            url.DisplayText = $"Dashboard ({result.Scheme})";
```

Comment:
Why not use the endpoint name?

Reply:
Done. I wasn't sure what happened when there were multiple http or https endpoints but it looks like they end up with a name matching there scheme.